lib/Util/SearchEndpointTrait.php

```diff
@@ -0,0 +1,167 @@
+     *
+     * @throws Exception
+     */
+    private function normalizeSearchVideosResponse($response)
+    {
+        $response = json_decode($response, true);
+        if (!is_array($response) || !array_key_exists('result', $response) || !array_key_exists('total', $response)) {
+            throw new Exception('Invalid response from search endpoint');
+        }
+
+        $response['totalCount'] = $response['total'];
+        $response['videos'] = $response['result'];
+        unset($response['result'], $response['total']);
+
+        foreach ($response['videos'] as $videoIndex => &$video) {
+            $video['uploadDate'] = $video['createdDate'];
+            $video['length'] = $video['duration'];
+            foreach ($video as $prop => $value) {
+                if (in_array($prop, ['createdDate', 'modifiedDate', 'uploadDate'])) {
+                    $video[$prop] = date_create($value)->getTimestamp();
+                }
+
+                if ($prop === 'channels') {
+                    foreach ($value as $channelIndex => $channelId) {
+                        $video[$prop][$channelIndex] = [
+                            'id' => $channelId,
+                            'name' => '',
+                        ];
+                    }
+                }
+            }
+        }
+
+        return json_encode($response);
+    }
+
+    /**
+     * Adjust the response from the `search` endpoint for channel data type.
+     * Namely, it renames the root-level properties, so they can be correctly unserialized.
+     *
+     * @param string $response
+     *
+     * @return string
+     *
+     * @throws Exception
+     */
+    private function normalizeSearchChannelsResponse($response)
+    {
+        $response = json_decode($response, true);
+        if (!is_array($response) || !array_key_exists('result', $response) || !array_key_exists('total', $response)) {
+            throw new Exception('Invalid response from search endpoint');
+        }
+
+        $response['totalCount'] = $response['total'];
+        $response['channels'] = $response['result'];
+        unset($response['result'], $response['total']);
+
+        return json_encode($response);
+    }
+
+    /**
+     * @param string $response
+     *
+     * @return int
+     *
+     * @throws Exception
+     */
+    private function getTotalCountFromSearchVideosResponse($response)
+    {
+        $response = json_decode($response, true);
+        if (!is_array($response) || !array_key_exists('total', $response)) {
+            throw new Exception('Response from search endpoint is missing the "total" key');
+        }
+
+        return (int) $response['total'];
+    }
+
+    /**
+     * @param int                          $videoManagerId
+     * @param VideosRequestParameters|null $parameters
+     *
+     * @return array
+     */
+    private function getRequestOptionsForSearchVideosEndpoint(
+        $videoManagerId,
+        VideosRequestParameters $parameters = null
```

Comment:
I think you have to use "?VideosRequestParameters"

Reply:
That is a PHP 7.1 feature. This library needs to support 5.6

Reply:
OK, you are right :)